egi_notebooks_accounting/prometheus.py

```diff
@@ -0,0 +1,270 @@
+#! /usr/bin/python3
+
+import argparse
+import json
+import logging
+import os
+import peewee
+import requests
+import sys
+import time
+import urllib3
+from configparser import ConfigParser
+from datetime import datetime
+from dirq import QueueSimple
+from requests.auth import HTTPBasicAuth
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+# urllib3 1.9.1: from urllib3.exceptions import InsecureRequestWarning
+
+from .model import VM, db, db_init
+
+
+CONFIG = 'prometheus'
+DEFAULT_CONFIG_FILE = 'config.ini'
+DEFAULT_FILTER = 'pod=~\'jupyter-.*\''
+DEFAULT_FQANS = {}
+DEFAULT_FQAN_KEY = 'primary_group'
+DEFAULT_PROMETHEUS_URL = 'http://localhost:8080'
+DEFAULT_RANGE = '4h'
+
+
+def log(msg):
+    logging.debug('%s' % msg)
+
+
+def warn(msg):
+    logging.warning('%s' % msg)
+
+
+class Prometheus:
+    DEFAULT_AGENT = 'egi-notebooks-client/1.0-dev'
+    DEFAULT_HEADERS = {'User-Agent': DEFAULT_AGENT}
+    DEFAULT_HEADERS_MIME = {'Content-Type': 'application/x-www-form-urlencoded'}
+
+    def __init__(self, parser):
+        config = parser[CONFIG] if CONFIG in parser else {}
+        url = os.environ.get('PROMETHEUS_URL', config.get('url', DEFAULT_PROMETHEUS_URL))
+        if not url.endswith('/'):
+            url = url + '/'
+        self.url = url + 'api/v1'
+        user = config.get('user')
+        password = config.get('password')
+        self.auth = HTTPBasicAuth(user, password)
+        self.headers = Prometheus.DEFAULT_HEADERS
+        self.verify = os.environ.get('SSL_VERIFY', config.get('verify', 1))
+        if self.verify != '1' and self.verify != 'True':
+            self.verify = 0
+        if not self.verify:
+            urllib3.disable_warnings(InsecureRequestWarning)
+        log('Site %s' % VM.site)
+        log('URL %s' % self.url)
+        log('verify %s' % self.verify)
+        self.pods = dict()
+
+    def handle_error(self, response):
+        response.raise_for_status()
+
+    def get(self, rel_url):
+        """REST GET request."""
+        url = self.url + rel_url
+        response = requests.get(url, auth=self.auth, verify=self.verify,
```

Comment:
maybe use a `session` from requests: https://requests.readthedocs.io/en/latest/user/advanced/#session-objects, so you can just have in `__init__` something like:

```
self.session = request.Session()
self.session.auth = ...
self.headers.update(...)
self.verify = ...
```

and here just
```
self.session.get(url)
```